pkg/connector/enterprise_applications.go

```diff
@@ -381,13 +378,12 @@ func (id *enterpriseApplicationsEntitlementId) MarshalString() (string, error) {
 func (id *enterpriseApplicationsEntitlementId) UnmarshalString(input string) error {
 	parts := strings.Split(input, ":")
 	if len(parts) < 3 {
-		return errors.New("baton-microsoft-entra: invalid entitlement id")
+		return errors.New("baton-azure-infrastructure: invalid entitlement id")
 	}
-	id.Type = parts[2]
-	id.Resource = parts[1]
+	id.Type = parts[1]
```

Comment:
type should still be `id.Type = parts[2]` because the entitlement id is in the structure of resource-type:resource-id:type

Reply:
My mistake, you were right.